src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -338,6 +338,7 @@ bool IStructuralEquatable.Equals(object other, IEqualityComparer comparer)
 
             var objTuple = (ValueTuple<T1>)other;
 
+            if (comparer == null) comparer = EqualityComparer<object>.Default;
             return comparer.Equals(Item1, objTuple.Item1);
```

Comment:
Do you want me to go ahead with this? I assume that the larger code size would not be a problem?